framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -435,22 +443,24 @@ private AnnotationMirror createBooleanAnnotationMirror(List<Boolean> values) {
 
     private List<Number> calculateNumericalUnaryOp(
             Node operand, NumericalUnaryOps op, TransferInput<CFValue, CFStore> p) {
-        List<? extends Number> lefts = getNumericalValues(operand, p);
         List<Number> resultValues = new ArrayList<>();
-        for (Number left : lefts) {
-            NumberMath<?> nmLeft = NumberMath.getNumberMath(left);
-            switch (op) {
-                case PLUS:
-                    resultValues.add(nmLeft.unaryPlus());
-                    break;
-                case MINUS:
-                    resultValues.add(nmLeft.unaryMinus());
-                    break;
-                case BITWISE_COMPLEMENT:
-                    resultValues.add(nmLeft.bitwiseComplement());
-                    break;
-                default:
-                    throw new UnsupportedOperationException();
+        if (!isIntRange(operand, p)) {
```

Comment:
You are right about this. In current PR, there is NO analysis support for expressions involve @IntRange annotation.

In future patches, I will try to address the following issues:
1. handle @IntRange only binary operations (e.g., @IntRange + @IntRange)
2. handle @IntRange mixed with @IntVal binary operations (as you suggested)
3. handle @IntRange unary operations (e.g., -@IntRange)

Besides, I am not sure in this PR if I should implement the control flow type refinement for comparison operations, e.g. 

     void foo(
          @IntRange(from = 0, to = 10) int _a, 
          @IntRange(from = 3, to = 5) int _b) 
          if (_a <= _b) {
              // range of _a is now refined to [0, 5] because a value in range [6, 10]
              // cannot be less than or equal to variable _b with range [3, 5]
              ...
          }
      }

This is not implemented in the original value checker.